apps/internal/oauth/ops/authority/authority.go

```diff
@@ -213,9 +214,23 @@ func NewInfoFromAuthorityURI(authorityURI string, validateAuthority bool) (Info,
 		return Info{}, err
 	}
 
+	port := u.Port()
```

Comment:
I would add a comment around this functionality, i.e. preserve the port number. This is a bug that can be fixes separately from this proposal.